src/include/network/connection_context.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include "traffic_cop/portal.h"
+#include "traffic_cop/statement.h"
+
+namespace terrier::network {
+
+/**
+ * A ConnectionContext stores the state of a connection.
+ */
+
+struct ConnectionContext {
+  /**
+   * The statements in this connection
+   */
+  std::unordered_map<std::string, traffic_cop::Statement> statements;
+
+  /**
+   * The portals in this connection
+   */
+  std::unordered_map<std::string, traffic_cop::Portal> portals;
+
+  /**
+   * Cleans up this ConnectionContext
+   */
+  void Reset() {
```

Comment:
Can you explain when should a `ConnectionContext` be reset? Is it when the connection closes or when the query finishes executing?

Reply:
Thanks for the comment! It's called when its connection handle is reused or destroyed. I have added doc on that.